### PR TITLE
Refactor channel state in lightningd

### PR DESCRIPTION
**Pull request by rustyrussell**
It started with our handling of the funding depth callback, which needed cleaning up badly.  But it lead me quickly to our channel state functions, which are unclear, especially as we added dual funding states and a splicing state.  This makes the code cleaner and easier to update in future.

I stopped short of converting "uncommitted_channel" to a channel in a new state, but that is the next obvious step (this entire series was a side-quest as I was trying to clean up our channel_update handling!).

### CODE REVIEW

File: lightningd/peer_control.c [hunk 1]
```diff
@@ -1833,18 +1833,50 @@ void update_channel_from_inflight(struct lightningd *ld,
...
+	case AWAITING_UNILATERAL:
+	case CHANNELD_SHUTTING_DOWN:
+	case CLOSINGD_SIGEXCHANGE:
+	case CLOSINGD_COMPLETE:
+	case FUNDING_SPEND_SEEN:
+	case ONCHAIN:
+	case CLOSED:
+	case DUALOPEND_OPEN_INIT:
+		return;
+
```
Comment (vincenzopalazzo) [inline lightningd/peer_control.c:1870]: nit: not needed space?

File: lightningd/peer_control.c [hunk 2]
```diff
@@ -1833,18 +1833,50 @@ void update_channel_from_inflight(struct lightningd *ld,
...
+	case CHANNELD_NORMAL:
+	case CHANNELD_AWAITING_LOCKIN:
+	case CHANNELD_AWAITING_SPLICE:
+		channeld_tell_depth(channel, txid, depth);
+		return;
+
```
Comment (vincenzopalazzo) [inline lightningd/peer_control.c:1876]: same here (?)

**Review by vincenzopalazzo — COMMENTED** [review]
Left some comments while I was review the PR to understand what was changed from before

In order to fix lnprototest you should build on top of https://github.com/ElementsProject/lightning/pull/6702


The problem was that lnprototest was counting the current block heigh in the wrong way and this mean that core lightning was going out of sync (causing the reorg)

**Comment by ddustin** [comment]
Oh sweet! A clean up of funding depth would be awesome!

**Comment by rustyrussell** [comment]
OK, this needed some rework: there was more to clean up!

In particular, we used funding_depth_cb for both the main funding tx, as well as inflights (DF RBF attempts, splicing). This was deeply confusing, and caused weird issues until I understood it. I've now separated those out.

I broke the splicing gossip test, but this is the basis for reworking gossip in general, so I will workaround that test for now.

**Pull request merged**